bigtable/client/rpc_backoff_policy.cc

```diff
@@ -46,12 +46,25 @@ void ExponentialBackoffPolicy::setup(grpc::ClientContext& /*unused*/) const {}
 std::chrono::milliseconds ExponentialBackoffPolicy::on_completion(
     grpc::Status const& status) {
   using namespace std::chrono;
-  // TODO(#57) - we also need to randomize the sleep period.
   auto delay = duration_cast<milliseconds>(current_delay_);
-  current_delay_ *= 2;
-  if (current_delay_ >= maximum_delay_) {
-    current_delay_ = maximum_delay_;
+  std::uniform_int_distribution<int> rng_distribution(0,
+                                                      current_delay_.count());
+  // Randomized sleep period because it is possible that after some time all
+  // client have same sleep period if we use only exponential backoff policy.
+  auto random_delay = microseconds(rng_distribution(generator_));
+  random_delay *= 2;
+  if (random_delay <= current_delay_) {
+    std::uniform_int_distribution<int> distribution(
+        random_delay.count(), 3 * (random_delay.count()) / 2);
+    random_delay = microseconds(distribution(generator_));
+    random_delay *= 2;
   }
+  if (current_delay_ > maximum_delay) {
+    std::uniform_int_distribution<int> distribution(
+        3 * maximum_delay.count() / 4, 5 * maximum_delay.count() / 4);
+    random_delay = microseconds(distribution(generator_));
```

Comment:
So in the worst case `random_delay` can be initialized to `5 * maximum_delay_ / 4`, which is higher than `maximum_delay_` and therefore the `current_delay_` will be set to a value larger than the configured maximum.

I think your code is overly complex, and it makes it easy to miss such things.  Is there a reason you do not like my proposal from earlier:

```C++
std::uniform_int_distribution<int> distribution(0, duration_cast<microseconds>(current_delay_).count());
auto delay = microseconds(distribution(generator));
current_delay_ *= 2;
if (current_delay_ >= maximum_delay_) {
  current_delay_ = maximum_delay_;
}
return delay;
```

Reply:
Fixed.